tasks/post_to_queue_and_trigger_step_function/post_to_queue_and_trigger_step_function.py

```diff
@@ -0,0 +1,128 @@
+"""
+Name: post_to_queue_and_trigger_step_function.py
+
+Description: Receives an events from an SQS queue, translates to get_current_archive_list's input format,
+sends it to another queue, then triggers the internal report step function.
+"""
+import json
+import os
+from typing import Any, Dict, TypeVar
+
+import boto3
+import fastjsonschema
+
+# noinspection PyPackageRequirements
+from cumulus_logger import CumulusLogger
+
+import sqs_library
+from sqs_library import retry_error
+
+OS_ENVIRON_TARGET_QUEUE_URL_KEY = "TARGET_QUEUE_URL"
+OS_ENVIRON_STEP_FUNCTION_ARN_KEY = "STEP_FUNCTION_ARN"
+
+OUTPUT_REPORT_BUCKET_REGION_KEY = "reportBucketRegion"
+OUTPUT_REPORT_BUCKET_NAME_KEY = "reportBucketName"
+OUTPUT_MANIFEST_KEY_KEY = "manifestKey"
+
+RT = TypeVar("RT")  # return type
+
+LOGGER = CumulusLogger(name="ORCA")
+# Generating schema validators can take time, so do it once and reuse.
+try:
+    with open("schemas/input.json", "r") as raw_schema:
+        _INPUT_VALIDATE = fastjsonschema.compile(json.loads(raw_schema.read()))
+except Exception as ex:
+    LOGGER.error(f"Could not build schema validator: {ex}")
+    raise
+
+try:
+    with open("schemas/body.json", "r") as raw_schema:
+        _BODY_VALIDATE = fastjsonschema.compile(json.loads(raw_schema.read()))
+except Exception as ex:
+    LOGGER.error(f"Could not build schema validator: {ex}")
+    raise
+
+
+def process_record(
+    record: Dict[str, Any],
+    target_queue_url: str,
+    step_function_arn: str,
+) -> None:
+    """
+    Central method for translating record to pass along and triggering step function.
+
+    Args:
+        record: The record to post.
+        target_queue_url: The url of the queue to post the records to.
+        step_function_arn: The arn of the step function to trigger.
+    """
+    new_body = translate_record_body(record["body"])
+    sqs_library.post_to_fifo_queue(target_queue_url, new_body)
```

Comment:
Per our discussion on FIFO and how a FIFO DLQ we think will break the issue where processing would stop because order must be maintained, will leave as FIFO for now until we see some real world use/testing.